client/state/data-layer/wpcom/sites/rewind/schema.js

```diff
@@ -0,0 +1,102 @@
+/** @format */
+
+export const credential = {
+	type: 'object',
+	properties: {
+		still_valid: { type: 'boolean' },
+		type: { type: 'string', enum: [ 'auto', 'ftp', 'sftp', 'ssh' ] },
+		host: { type: 'string' },
+		port: { type: 'integer' },
+	},
+	required: [ 'still_valid', 'type' ],
+};
+
+export const download = {
+	type: 'object',
+	properties: {
+		downloadId: { type: 'integer' },
+		rewindId: { type: 'string' },
+		backupPoint: { type: 'integer' },
+		startedAt: { type: 'integer' },
+		downloadCount: { type: 'integer' },
+		validUntil: { type: 'integer' },
+	},
+	required: [ 'downloadId', 'rewindId', 'backupPoint', 'startedAt' ],
+};
+
+export const restore = {
+	type: 'object',
+	properties: {
+		status: { type: 'string', enum: [ 'finished', 'inactive', 'queued', 'running' ] },
+		percent: { type: 'integer' },
+	},
+	required: [ 'status' ],
+};
+
+export const unavailable = {
+	type: 'object',
+	properties: {
+		state: {
+			type: 'string',
+			pattern: '^unavailable$',
+		},
+		reason: {
+			type: 'string',
+		},
+		last_updated: { type: 'integer' },
+	},
+	required: [ 'state', 'last_updated' ],
+};
+
+export const inactive = {
+	type: 'object',
+	properties: {
+		state: {
+			type: 'string',
+			pattern: '^inactive$',
+		},
+		credentials: {
+			type: 'array',
+			items: credential,
+		},
+		last_updated: { type: 'integer' },
+	},
+	required: [ 'state', 'last_updated' ],
+};
+
+export const awaitingCredentials = {
```

Comment:
I think we may want to consider renaming this to something slightly more descriptive of what the state actually is. `awaitingCredentials` might be confused with another state we need to work into the machine - when credentials have been requested, but are not yet actually provisioned (a job is running to provision). Perhaps we can call this something like `awaitingCredentialsInput`?